llvm/lib/Target/AArch64/AArch64InstrInfo.td

```diff
@@ -6068,6 +6068,32 @@ def : Pat<(f16 (any_sint_to_fp (i32 (any_fp_to_sint f16:$Rn)))),
 def : Pat<(f16 (any_uint_to_fp (i32 (any_fp_to_uint f16:$Rn)))),
           (UCVTFv1i16 (f16 (FCVTZUv1f16 f16:$Rn)))>;
 }
+
+// int -> float conversion of value in lane 0 of simd vector should use 
+// correct cvtf variant to avoid costly fpr <-> gpr register transfers.
+def : Pat<(f32 (sint_to_fp (i32 (vector_extract (v4i32 FPR128:$Rn), (i64 0))))),
```

Comment:
Do these need a HasNEONandIsStreamingSafe predicate?

Reply:
I believe so - the entire section that these patterns are in is predicated by a HasNeonandIsStreamingSafe slightly further up, though admittedly this is not that clear.

Reply:
I agree with @davemgreen. It looks like these patterns need guarding with `let Predicates = [HasNEONandIsStreamingSafe]` similar to the two fp16 patterns below.

Reply:
Apologies if my comment was unclear, these patterns already fall in the scope of a previous `let Predicates = [HasNEONandIsStreamingSafe]` binding. I can always another guard above these patterns if we'd like this to be clearer?

Reply:
Ah I see. I mis-read where the above predicate got closed.